composer/algorithms/progressive_resizing/progressive_resizing.py

```diff
@@ -44,42 +46,61 @@ def resize_batch(X: torch.Tensor,
     Args:
         X: input tensor of shape (N, C, H, W). Resizing will be done along
             dimensions H and W using the constant factor ``scale_factor``.
-        y: output tensor of shape (N, C, H, W) that will also be resized if
+        y: output tensor of shape (N, H, W) or (N, C, H, W) that will also be resized if
             ``resize_targets`` is ``True``,
         scale_factor: scaling coefficient for the height and width of the
             input/output tensor. 1.0 keeps the original size.
         mode: type of scaling to perform. Value must be one of ``'crop'`` or
             ``'resize'``. ``'crop'`` performs a random crop, whereas ``'resize'``
-            performs a bilinear interpolation.
+            performs a nearest neighbor interpolation.
         resize_targets: whether to resize the targets, ``y``, as well
 
     Returns:
         X_sized: resized input tensor of shape ``(N, C, H * scale_factor, W * scale_factor)``.
         y_sized: if ``resized_targets`` is ``True``, resized output tensor
-            of shape ``(N, C, H * scale_factor, W * scale_factor)``. Otherwise
-            returns original ``y``.
+            of shape ``(N, H * scale_factor, W * scale_factor)`` or  ``(N, C, H * scale_factor, W * scale_factor)``.
+            Depending on the input ``y``. Otherwise returns original ``y``.
     """
     # Short-circuit if nothing should be done
     if scale_factor >= 1:
         return X, y
 
-    def resize_tensor(tensor: torch.Tensor) -> torch.Tensor:
-        # Reduce the size of input images, either via cropping or downsampling
-        if mode.lower() == "crop":
-            Hc = int(scale_factor * tensor.shape[2])
-            Wc = int(scale_factor * tensor.shape[3])
-            resize_transform = transforms.RandomCrop((Hc, Wc))
-        elif mode.lower() == "resize":
-            resize_transform = partial(F.interpolate, scale_factor=scale_factor, mode='bilinear')
-        else:
-            raise ValueError(f"Progressive mode '{mode}' not supported.")
-        return resize_transform(tensor)
-
-    X_sized = resize_tensor(X)
-    if resize_targets:
-        y_sized = resize_tensor(y)
+    # Prep targets for resizing if necessary
+    if check_for_index_targets(y) and resize_targets is True:
+        # Add a dimension to match shape of the input and change type for resizing
+        y_sized = y.float().unsqueeze(1)
     else:
         y_sized = y
+
+    if mode.lower() == "crop" and resize_targets is False:
+        # Make a crop transform for X
+        resize_transform = _make_crop(tensor=X, scale_factor=scale_factor)
+        X_sized, y_sized = resize_transform(X), y
```

Comment:
```suggestion
        X_sized = resize_transform(X)
```
I'm finding the logic behind when you use `y` vs `y_sized` a little complicated and unnecessary.

Reply:
This was done so that I could do a single `return X_sized, y_sized` at the end rather than multiple, and I found it easier to read the different parts of the if statement if they could each be read as applying a transform to X, y, which is sometimes the identity transform. Open to other suggestions here though.